web/src/main/java/org/springframework/security/web/access/expression/DefaultWebSecurityExpressionHandler.java

```diff
@@ -33,18 +32,16 @@
 public class DefaultWebSecurityExpressionHandler extends AbstractSecurityExpressionHandler<FilterInvocation>
 		implements SecurityExpressionHandler<FilterInvocation> {
 
-	private AuthenticationTrustResolver trustResolver = new AuthenticationTrustResolverImpl();
-
-	private String defaultRolePrefix = "ROLE_";
-
 	@Override
 	protected SecurityExpressionOperations createSecurityExpressionRoot(Authentication authentication,
 			FilterInvocation fi) {
 		WebSecurityExpressionRoot root = new WebSecurityExpressionRoot(authentication, fi);
 		root.setPermissionEvaluator(getPermissionEvaluator());
-		root.setTrustResolver(this.trustResolver);
-		root.setRoleHierarchy(getRoleHierarchy());
-		root.setDefaultRolePrefix(this.defaultRolePrefix);
+		// Since WebSecurityExpressionRoot is of a different type, use the default factory
+		// instead of the one configured with this class.
+		root.setTrustResolver(getDefaultAuthorizationManagerFactory().getTrustResolver());
+		root.setRoleHierarchy(getDefaultAuthorizationManagerFactory().getRoleHierarchy());
+		root.setDefaultRolePrefix(getDefaultAuthorizationManagerFactory().getRolePrefix());
```

Comment:
This is a similar problem to the comment for `DefaultMessageSecurityExpressionHandler`.

To solve it, we should create a package scoped `FilterInvocationContextRoot` that has a `public final HttpServletRequest request` member and extends `SecurityExpressionRoot<FilterInvocation>`.  This method would then use `FilterInvocationContextRoot` instead of `WebSecurityExpressionRoot`.

Then, we should create a ticket to consider deprecating the authorization logic that depends on `FilterInvocation` (e.g. the SecurityExressionHandler<FilterInvocation> and `FilterInvocationContextRoot`)